src/test/kotlin/com/msd/robot/application/RobotControllerTest.kt

```diff
@@ -103,4 +108,49 @@ class RobotControllerTest(
 
         assert(robotRepository.existsById(resultRobot.id))
     }
+
+    @Test
+    fun `Sending EnergyRegen Command with invalid robot UUID returns 400`() {
+        val command = "regenerate invalidRobotId ${UUID.randomUUID()}"
+        // when
+        mockMvc.post("/commands") {
+            contentType = MediaType.APPLICATION_JSON
+            content = mapper.writeValueAsString(CommandDTO(listOf(command)))
+        }.andExpect {
+            status { isBadRequest() }
+        }
+    }
+
+    @Test
+    fun `Sending EnergyRegen Command with invalid transaction UUID returns 400`() {
+        val command = "regenerate ${UUID.randomUUID()} invalidTransactionId"
```

Comment:
Create a Robot an assert that it doesn't get its energy restored.

Reply:
This test now fails. You have to actually persist the robot after using it. Also please verify that all tests work before pushing.